fondant/component.py

```diff
@@ -74,27 +74,37 @@ def _load_or_create_manifest(self) -> Manifest:
         """Abstract method that returns the dataset manifest"""
 
     @abstractmethod
-    def _process_dataset(self, dataset: FondantDataset) -> FondantDataset:
+    def _process_dataset(self, dataset: FondantDataset) -> dd.DataFrame:
         """Abstract method that processes the input dataframe and updates the Fondant Dataset
         with the new or loaded subsets"""
 
     def run(self):
```

Comment:
Would be nice to have a long descriptive docstring describing in-details what the `run()` method is responsible for. Alternatively, you can include this doc at the class definition level (`FondantComponent(ABC)`) to provide a more comprehensive explanation of the component's functionality.